src/core/bonded_coulomb_p3m_sr.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+  Copyright (C) 2010,2012,2013,2014,2015,2016 The ESPResSo project
+  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010 
+    Max-Planck-Institute for Polymer Research, Theory Group
+  
+  This file is part of ESPResSo.
+  
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+#ifndef _BONDED_COULOMB_P3M_SR_HPP
+#define _BONDED_COULOMB_P3M_SR_HPP
+/** \file bonded_coulomb.hpp
+ *  Routines to calculate the BONDED_COULOMB_P3M_SR Energy or/and BONDED_COULOMB_P3M_SR force 
+ *  for a particle pair. This is only the shortrange part of P3M and first 
+ *  used to subtract certain intramolecular interactions in combination with thole damping
+ *  \ref forces.cpp
+*/
+
+/************************************************************/
+
+#include "config.hpp"
+
+#ifdef P3M
+
+#include "utils.hpp"
+#include "debug.hpp"
+#include "interaction_data.hpp"
+#include "particle_data.hpp"
+#include "p3m.hpp"
+
+/// set the parameters for the bonded_coulomb potential
+int bonded_coulomb_p3m_sr_set_params(int bond_type, double q1q2);
+
+/** Computes the BONDED_COULOMB_P3M_SR pair force and adds this
+    force to the particle forces (see \ref interaction_data.cpp). 
+    @param p1        Pointer to first particle.
+    @param p2        Pointer to second/middle particle.
+    @param iaparams  bond parameters.
+    @param dx        particle distance vector
+    @param force     returns force of particle 1
+    @return 0.
+*/
+inline int calc_bonded_coulomb_p3m_sr_pair_force(Particle *p1, Particle *p2, Bonded_ia_parameters *iaparams, double dx[3], double force[3])
```

Comment:
avoid output parameter

Reply:
This is the standard signature for bonded ia. Please leave as it is.